cfg/builder/builder_walk.cc

```diff
@@ -385,8 +392,10 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
                 auto elseBlock = cctx.inWhat.freshBlock(cctx.loops);
                 conditionalJump(cont, ifSym, thenBlock, elseBlock, cctx.inWhat, a.cond.loc());
 
-                auto thenEnd = walk(cctx, a.thenp, thenBlock);
-                auto elseEnd = walk(cctx, a.elsep, elseBlock);
+                auto thenEnd = ast::isa_tree<ast::EmptyTree>(a.thenp) ? walkEmptyTreeInIf(cctx, a.loc, thenBlock)
+                                                                      : walk(cctx, a.thenp, thenBlock);
```

Comment:
I wonder if it's worth pushing this into `walkEmptyTreeInIf`, and calling it something like `walkConditionalBranch` instead?

Reply:
Yeah possibly, I'm a little hesitant to abstract prematurely, I think that we can possibly do something about this whenever we get around to ENFORCEing that we don't `walk` to `EmptyTree`.